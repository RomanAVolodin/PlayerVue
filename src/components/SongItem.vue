<template>
  <li :class="`song-id-${song.docId}`"
    class="snapshot example flex justify-between
    items-center p-3 pl-6 cursor-pointer transition duration-300 hover:bg-gray-50">
    <div>
      <router-link
        :to="{ name: 'song', params: { id: song.docId } }"
      >
        <span class="font-bold block text-gray-600 composition-name">
          {{ song.modified_name }}
        </span>
      </router-link>
      <span class="text-gray-500 text-sm">
        {{ song.genre }}
      </span>
    </div>

    <div class="text-gray-600 text-lg">
      <router-link
        custom
        :to="{ name: 'song', params: { id: song.docId }, hash: '#comments' }"
        v-slot="{ navigate }"
      >
        <span
          class="comments"
          @click="navigate"
        >
          <i class="fa fa-comments text-gray-600"></i>
          {{ song.comment_count }}
        </span>
      </router-link>
    </div>
  </li>
</template>
<script>
export default {
  name: 'SongItem',
  props: {
    song: {
      type: Object,
      required: true,
    },
  },
};
</script>
