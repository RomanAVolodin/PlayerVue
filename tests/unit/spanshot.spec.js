import { shallowMount, RouterLinkStub } from '@vue/test-utils';
import SongItem from '@/components/SongItem.vue';

describe('Snapshots SongItem.vue', () => {
  it('renders correctly', () => {
    const song = {
      docId: 'abc',
      modified_name: 'test',
      display_name: 'test',
      commenst_count: 5,
    };
    const wrapper = shallowMount(SongItem, {
      propsData: {
        song,
      },
      global: {
        components: {
          'router-link': RouterLinkStub,
        },
      },
    });

    expect(wrapper.element).toMatchSnapshot();
  });
});
