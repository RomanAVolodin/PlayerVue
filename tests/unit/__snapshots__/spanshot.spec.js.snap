// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshots SongItem.vue renders correctly 1`] = `
<li
  class="song-id-abc snapshot example flex justify-between items-center p-3 pl-6 cursor-pointer transition duration-300 hover:bg-gray-50"
>
  <div>
    <router-link-stub
      to="[object Object]"
    />
    <span
      class="text-gray-500 text-sm"
    />
  </div>
  <div
    class="text-gray-600 text-lg"
  >
    <router-link-stub
      custom=""
      to="[object Object]"
    />
  </div>
</li>
`;
